### PR TITLE
Update documentation

### PR DESCRIPTION
**Pull request by cassava**
New:
- Enable myst_parser extensions for Sphinx markdown
- Document cloe-engine hooks
- Document cloe-engine variable interpolation

Changed:
- Move doc on actions to developer's guide
- Complete documentation on `command` action

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -176,6 +176,12 @@ this shouldn't need to be performed manually.
 
 See the documentation on testing Cloe for more details.
 
+### Disabling Unit Tests
+
+When using `make` to build the project, add this to the command line:
```
Comment (clonker) [inline README.md:181]: How about adding a sentence on how to do it with native conan? Not that there's much of a difference anyways, but still.

File: docs/reference/variables.rst [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+The default string can also contain a variable.
+For example, the registry path is set like so by default::
+
+    ${XDG_DATA_HOME-${HOME}/.local/share}/cloe/registry
+
+There are several special variables that are set by Cloe engine:
+
+``THIS_STACKFILE_DIR``
+    Full path to directory containing stack file containing this variable.
```
[the remark this answers is not in the document]
Comment (cassava) [reply docs/reference/variables.rst:17]: lol, good suggestion!

File: docs/reference/variables.rst [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    }
+
+However, care should be taken to ensure that the stack file is valid
+JSON after interpolation. In the above example, if ``ENGINE_CONFIG`` is
+empty, then the JSON is invalid, because of the trailing comma.
+
+This approach does allow us to use variables in places where strings
+are not useable, such as for integers or booleans.
+
+Interpolation can be disabled or enabled from the command line,
```
[the remark this answers is not in the document]
Comment (cassava) [reply docs/reference/variables.rst:51]: Enabled, I think. Should add it.

File: docs/reference/hooks.rst [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+      "hooks": {
+        "pre_connect": [
+          { "path": "${VTD_LAUNCH}", "args": [ "stop" ] },
+          {
+            "path": "${VTD_LAUNCH}",
+            "args": [
+              "-l", "/tmp/vtd.log",
+              "start",
+              "--vtd-setup-cloe", "Cloe.noGUInoIG",
+              "--vtd-project", "${THIS_STACKFILE_DIR}/../contrib/projects/cloe_tests"
```
[the remark this answers is not in the document]
Comment (cassava) [reply docs/reference/hooks.rst:36]: Agreed, will simplify.

File: docs/reference/variables.rst (hunk 3 above)
[the remark this answers is not in the document]
Comment (cassava) [reply docs/reference/variables.rst:51]: It's actually specified already. The reason I write both is, I think you can have some alias that sets `--no-interpolate` and then write `--interpolate` for the cases you want it.

File: docs/reference/variables.rst [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+There are several special variables that are set by Cloe engine:
+
+``THIS_STACKFILE_DIR``
+    Full path to directory containing stack file containing this variable.
+
+``THIS_STACKFILE_FILE``
+    Full path to stack file containing this variable.
+
+``CLOE_SIMULATION_UUID``
+    If this variable is not
```
[the remark this answers is not in the document]
Comment (cassava) [reply docs/reference/variables.rst:23]: Yup, missed that one...

**Review by clonker — APPROVED** [review]

**Pull request merged**